connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -307,15 +307,21 @@ private void processConnectorConfigUpdates(Set<String> connectorConfigUpdates) {
     }
 
     private void processTargetStateChanges(Set<String> connectorTargetStateChanges) {
-        if (!connectorTargetStateChanges.isEmpty()) {
-            for (String connector : connectorTargetStateChanges) {
-                if (worker.connectorNames().contains(connector)) {
-                    TargetState targetState = configState.targetState(connector);
-                    worker.setTargetState(connector, targetState);
-                    if (targetState == TargetState.STARTED)
-                        reconfigureConnectorTasksWithRetry(connector);
-                }
+        for (String connector : connectorTargetStateChanges) {
+            TargetState targetState = configState.targetState(connector);
+            if (targetState == null) {
+                log.warn("Skipping connector {} state change since we could not determine the target state", connector);
```

Comment:
Why would this happen? Is this the case for connector deletion? It seems like KafkaConfigBackingStore will write the null value via removeConnectorConfig, but I think the ConsumeCallback may not process it correctly -- we check for a null value, but don't seem to remove/write the null in that case, so I'm not sure it can propagate here (although the handling here may be fine, I guess that might be a bug in KafkaConfigBackingStore).

And now I see after reviewing more that this seems to be for the case of an unknown connector, but may have revealed a bug for deleted connectors too...